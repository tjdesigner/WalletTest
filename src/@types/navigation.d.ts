/**
 * Learn more about using TypeScript with React Navigation:
 * https://reactnavigation.org/docs/typescript/
 */

import { ParamListBase } from "@react-navigation/native"
import { NativeStackScreenProps } from "@react-navigation/native-stack"

import type {
  CompositeScreenProps,
  NavigatorScreenParams,
} from "@react-navigation/native"
import type { StackScreenProps } from "@react-navigation/stack"

interface Card {
  id?: string
  cardNumber: string
  name: string
  expirationDate: string
  cvv: string
}

/****************************************************/

export type RootStackParamList = {
  Home: Card | undefined

  Register:
    | {
        card?: Card
      }
    | undefined

  RegisterConfirmation:
    | {
        card: Card
      }
    | undefined

  LoadingAnimation: undefined

  MyCards:
    | {
        allCards?: Card[]
      }
    | undefined
}

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}

export type RootStackScreenProps<ScreenName extends keyof RootStackParamList> =
  NativeStackScreenProps<RootStackParamList, ScreenName>
