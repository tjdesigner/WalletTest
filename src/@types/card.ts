export interface Card {
  id?: string
  cardName?: string
  cardNumber?: string
  name?: string
  expirationDate?: string
  cvv?: string
}
