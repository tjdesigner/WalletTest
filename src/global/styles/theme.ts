export default {
  colors: {
    black: "black",
    white: "white",
    danger: "#DC1637",
    info: "#12C2E9",
    warning: "#FFE900",
    success: "#2ec64e",
    background: "rgba(238, 238, 238, .2)",
    primary: "#142995",
    secondary: "#12C2E9",
    tertiary: "#A5FF32",
    backgroundDisabled: "#EEEEEE",
    textDisabled: "#BBBBBB",
  },

  fonts: {
    family: {
      regular: "PTSans-Regular",
      italic: "PTSans-Italic",
      boldItalic: "PTSans-BoldItalic",
      bold: "PTSans-Bold",
    },
    weight: {
      normal: "normal",
      bold: "bold",
      l1: "100",
      l2: "200",
      l3: "300",
      l4: "400",
      l5: "500",
      l6: "600",
      l7: "700",
      l8: "800",
      l9: "900",
      undefined,
    },
  },
  fontSize: {
    xxxs: `${1}px`,
    xxs: `${2}px`,
    xs: `${4}px`,
    small: `${8}px`,
    default: `${16}px`,
    medium: `${24}px`,
    large: `${32}px`,
    large2: `${36}px`,
    large3: `${40}px`,
    xLarge: `${44}px`,
    xxLarge: `${48}px`,
    xxxLarge: `${60}px`,
    veryLarge: `${76}px`,
    titlePage: `${20}px`,
    title: `${28}px`,
    specific: `${18}px`,
  },
  fontSizeNumber: {
    xxxs: 1,
    xxs: 2,
    xs: 4,
    small: 8,
    default: 16,
    medium: 24,
    large: 32,
    xLarge: 44,
    xxLarge: 48,
    xxxLarge: 60,
    veryLarge: 76,
    titlePage: 20,
    title: 28,
    specific: 18,
  },
  radius: {
    xxs: `${2}px`,
    xs: `${4}px`,
    small: `${8}px`,
    small2: `${12}px`,
    default: `${16}px`,
    medium: `${20}px`,
    medium2: `${24}px`,
    medium3: `${28}px`,
    large: `${32}px`,
    large2: `${36}px`,
    large3: `${40}px`,
    xLarge: `${44}px`,
    xxLarge: `${48}px`,
    xxxLarge: `${54}px`,
    veryLarge: `${76}px`,
  },
  radiusNumber: {
    xxs: 2,
    xs: 4,
    small: 8,
    small2: 12,
    default: 16,
    medium: 24,
    large: 32,
    large2: 36,
    large3: 40,
    xLarge: 44,
    xxLarge: 48,
    xxxLarge: 60,
    veryLarge: 76,
  },
  spaces: {
    xxs: `${2}px`,
    xs: `${4}px`,
    small: `${8}px`,
    default: `${16}px`,
    medium: `${20}px`,
    medium2: `${24}px`,
    medium3: `${28}px`,
    large: `${32}px`,
    large2: `${36}px`,
    large3: `${40}px`,
    xLarge: `${44}px`,
    xxLarge: `${48}px`,
    xxxLarge: `${54}px`,
    veryLarge: `${76}px`,
  },
  spacesNumber: {
    xxs: 2,
    xs: 4,
    small: 8,
    default: 16,
    medium: 24,
    large: 32,
    large2: 36,
    large3: 40,
    xLarge: 44,
    xxLarge: 48,
    xxxLarge: 60,
    veryLarge: 76,
  },
}
