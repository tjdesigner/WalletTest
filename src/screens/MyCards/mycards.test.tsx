import React, { useState as useStateMock } from "react";
import { act, fireEvent, render } from '@testing-library/react-native'
import { RootStackParamList } from "../../@types/navigation";
import type { NativeStackScreenProps } from "@react-navigation/native-stack";
import { ThemeProvider } from "styled-components";
import theme from "../../global/styles/theme";
import MyCardsScreen from ".";
import dataBase from '../../../db.json'
import { Alert } from 'react-native';

const navigateMock = jest.fn()

jest.mock('uuid', () => (
    {
        v4: () => '1234asdf'
    }
))

jest.mock('../../service/api', () => {
    const database = require('../../../db.json')
    return ({
        getApiData: () => database
    })
})

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: () => jest.fn()
}));

jest.mock('react-native-reanimated', () =>
    require('react-native-reanimated/mock'),
);

jest.mock('react-native', () => (
    {
        alert: () => jest.fn()
    }
));


const mockOnPress = jest.fn()

type MyCardScreenProps = NativeStackScreenProps<RootStackParamList, 'MyCards'>

const optionsNavigation = {
    navigation: {
        navigate: navigateMock
    }
} as unknown as MyCardScreenProps

describe('MyCards', () => {

    const setState = jest.fn()
    //beforeEach(() => {
    // ; (useStateMock as jest.Mock).mockImplementation(init => [init, setState])
    //useStateMock.mockImplementation((init: any) => [dataBase.cards, setState])
    //})

    it('should render mycards screen', () => {
        useStateMock.mockImplementation((init: any) => [dataBase.cards, setState])
        const { toJSON } = render(<ThemeProvider theme={theme}>
            <MyCardsScreen {...optionsNavigation} />
        </ThemeProvider>)


        expect(toJSON()).toMatchSnapshot()
        // act(() => {
        //     setState(database)
        // });
    })

    it('should click button select card', async () => {
        const spyAlert = jest.spyOn(Alert, 'alert')
            //@ts-ignore
            .mockImplementation((title, message, callbackOrButtons) => callbackOrButtons[0].onPress())


        useStateMock.mockImplementation((init: any) => [dataBase.cards, setState])
        const { findByTestId, getByTestId, findByText } = render(<ThemeProvider theme={theme}>
            <MyCardsScreen {...optionsNavigation} />
        </ThemeProvider>)

        const myCardButton = await findByTestId('select-card-testid')
        fireEvent.press(myCardButton)
        fireEvent.press(await findByText('Entendi'));

        expect(Alert.alert).toHaveBeenCalled()
    })

    it('Mocking Alert', () => {
        const wrapper = shallow(<Search />);
        wrapper.findWhere(n => n.props().title == 'Submit').simulate('Press');
        expect(Alert.alert).toHaveBeenCalled(); // passes
        Alert.alert.mock.calls[0][2][0].onPress() // trigger the function within the array
        expect(wrapper.state('someState')).toBe(true)
    })

    it('should render mycards screen', () => {
        useStateMock.mockImplementation((init: any) => [[], setState])
        const { toJSON } = render(<ThemeProvider theme={theme}>
            <MyCardsScreen {...optionsNavigation} />
        </ThemeProvider>)
        expect(toJSON()).toMatchSnapshot()
    })

    it('should click button add first card', async () => {
        useStateMock.mockImplementation((init: any) => [[], setState])
        const { findByTestId } = render(<ThemeProvider theme={theme}>
            <MyCardsScreen {...optionsNavigation} />
        </ThemeProvider>)

        const myCardButton = await findByTestId('add-fist-card-testid')
        fireEvent.press(myCardButton)

        expect(navigateMock).toHaveBeenCalled()
    })
})
