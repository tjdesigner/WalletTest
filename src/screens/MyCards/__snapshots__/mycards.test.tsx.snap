// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCards should render mycards screen 1`] = `
[
  <View
    style={
      {
        "backgroundColor": "#142995",
        "marginTop": 100,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "borderBottomLeftRadius": 40,
          "borderBottomRightRadius": 40,
          "height": 54,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <Text
        color="#12C2E9"
        style={
          {
            "color": "#12C2E9",
            "fontFamily": "PTSans-Regular",
            "fontSize": 20,
            "marginBottom": 0,
          }
        }
        text="Meus cartões"
        type="topPage"
      >
        Meus cartões
      </Text>
    </View>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#142995",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      onTouchEnd={[Function]}
      style={
        [
          {
            "borderRadius": 16,
            "height": 180,
            "marginTop": 160.08,
            "position": "absolute",
            "width": 690,
            "zIndex": 3,
          },
          {
            "backgroundColor": "black",
            "top": -10,
            "transform": [
              {
                "scale": 1,
              },
              {
                "translateY": 170,
              },
            ],
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": ""flex-start"",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "white",
                "marginBottom": 32,
              },
            ]
          }
        >
          Black Card
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "white",
              },
            ]
          }
        >
          Tiago de Jesus Sousa Gama
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "white",
              },
            ]
          }
        >
          **** **** **** 1234
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "white",
              },
            ]
          }
        >
          Validade 12/32
        </Text>
      </View>
    </View>
    <View
      onTouchEnd={[Function]}
      style={
        [
          {
            "borderRadius": 16,
            "height": 180,
            "marginTop": 100.08000000000001,
            "position": "absolute",
            "width": 690,
            "zIndex": 2,
          },
          {
            "backgroundColor": "#A5FF32",
            "top": -10,
            "transform": [
              {
                "scale": 1,
              },
              {
                "translateY": 170,
              },
            ],
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": ""flex-start"",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "black",
                "marginBottom": 32,
              },
            ]
          }
        >
          Green Card
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "black",
              },
            ]
          }
        >
          Asdf sdf asfasf  
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "black",
              },
            ]
          }
        >
          **** **** **** 1234
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "black",
              },
            ]
          }
        >
          Validade 12/31
        </Text>
      </View>
    </View>
    <View
      onTouchEnd={[Function]}
      style={
        [
          {
            "borderRadius": 16,
            "height": 180,
            "marginTop": 40.08000000000001,
            "position": "absolute",
            "width": 690,
            "zIndex": 1,
          },
          {
            "backgroundColor": "black",
            "top": -10,
            "transform": [
              {
                "scale": 1,
              },
              {
                "translateY": 170,
              },
            ],
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": ""flex-start"",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "white",
                "marginBottom": 32,
              },
            ]
          }
        >
          Black Card
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "white",
              },
            ]
          }
        >
          Asdf asdf 1
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "white",
              },
            ]
          }
        >
          **** **** **** 4234
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "PTSans-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
              },
              {
                "color": "white",
              },
            ]
          }
        >
          Validade 23/12
        </Text>
      </View>
    </View>
    <View
      alignItems="center"
      backgroundColor="primary"
      marginTop={933.8}
      paddingLeft={32}
      paddingRight={32}
      style={
        {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#142995",
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 933.8,
          "paddingBottom": 0,
          "paddingLeft": 32,
          "paddingRight": 32,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#12C2E9",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "height": 54,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="select-card-testid"
      >
        <Text
          activeOpacity={0.7}
          backgroundColor="secondary"
          color="white"
          fullWidth={true}
          onPress={[Function]}
          style={
            {
              "backgroundColor": "#12C2E9",
              "color": "white",
              "fontFamily": "PTSans-Regular",
              "fontSize": 18,
              "fontWeight": "100",
            }
          }
          suppressHighlighting={true}
          testIDButton="select-card-testid"
          textButton="pagar com esse cartão"
        >
          pagar com esse cartão
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`MyCards should render mycards screen 2`] = `
[
  <View
    style={
      {
        "backgroundColor": "#142995",
        "marginTop": 100,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "borderBottomLeftRadius": 40,
          "borderBottomRightRadius": 40,
          "height": 54,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <Text
        color="#12C2E9"
        style={
          {
            "color": "#12C2E9",
            "fontFamily": "PTSans-Regular",
            "fontSize": 20,
            "marginBottom": 0,
          }
        }
        text="Meus cartões"
        type="topPage"
      >
        Meus cartões
      </Text>
    </View>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#142995",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      alignItems="center"
      backgroundColor="primary"
      marginTop={466.9}
      paddingLeft={32}
      paddingRight={32}
      style={
        {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#142995",
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 466.9,
          "paddingBottom": 0,
          "paddingLeft": 32,
          "paddingRight": 32,
          "paddingTop": 0,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "marginBottom": 32,
          }
        }
      >
        Você ainda não adicionou cartão.
      </Text>
    </View>
    <View
      alignItems="center"
      backgroundColor="primary"
      marginTop={400.2}
      paddingLeft={32}
      paddingRight={32}
      style={
        {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#142995",
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 400.2,
          "paddingBottom": 0,
          "paddingLeft": 32,
          "paddingRight": 32,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "white",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "height": 54,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="add-fist-card-testid"
      >
        <Text
          activeOpacity={0.7}
          backgroundColor="white"
          color="primary"
          fullWidth={true}
          onPress={[Function]}
          style={
            {
              "backgroundColor": "white",
              "color": "#142995",
              "fontFamily": "PTSans-Regular",
              "fontSize": 18,
              "fontWeight": "100",
            }
          }
          suppressHighlighting={true}
          testIDButton="add-fist-card-testid"
          textButton="Adicionar cartão"
        >
          Adicionar cartão
        </Text>
      </View>
    </View>
  </View>,
]
`;
